### ISSUE TITLE
Tool is resulting in problematic edits being made to OpenStreetMap

### ISSUE THREAD

**Issue by berrely** [chadrockey/TGC-Designer-Tools#94]
Hi,
I'm a volunteer contributor to the geographical dataset OpenStreetMap, which appears to be used by this project.

OpenStreetMap is used by many navigational apps such as Apple Maps and TomTom, and for humanitarian efforts. However, recently, due to videos like [this](https://youtu.be/WYa4ODgo-Bc), new users ("golf mappers") have been deliberately modifying data such as secondary roads, which are used for routing, incorrectly marking golf structures as buildings, and other deliberately incorrect edits with the aim of making them render nicely in this dataset.

This is creating problems for other programs that make use of OpenStreetMap data, and requires contributors to dedicate hours of their time to reverting and cleaning up such edits (which are still being found today). It's breaking the integrity and accuracy of the data, which creates issues for everyone.

The reason for this incorrect tagging and deletions seem to be predominantly due to shortcomings in the data processor used by TGCDT, such as it's rendering of roads. It seems best that the tags used to render objects from OSM data should be double checked and validated.

**Comment by chadrockey** [comment]
Sorry, that video with the bad editing practices already seems to have been removed so I can’t view for sure what exactly the editors were doing.

The video game this project is based on actually can’t render roads at all.  If they are imported into the game it’s only intended for guidelines for editing not as any finished product.

Im unable to modify or “un-sign” the distributed versions of this software.  If I were to make any changes, it’s likely users won’t download new versions or would intentionally use the older versions.

I’m sorry the untrained users are performing incorrect edits.  Mostly, they are outside of my influence and are non technical users.  I’d recommend reverting all of their edits and giving them a warning on first offense, then reverting all edits and IP banning on a second offense.

**Comment by berrely** [comment]
Thanks for explaining. At least you're aware :) Have a good week.

**Comment by Woazboat** [comment]
I don't quite understand your line of reasoning here. Users are modifying OSM data in unintended ways to work around the lack of support in this software in order to get them recognized. If you release a new version of the software that does what users want without having to do more work (vandalizing OSM along the way), why would they intentionally go out of their way to download and use an old version of the software?

(I'm not sure how cart paths are used here in this project, but obviously they serve some kind of purpose that users want. Otherwise why would people go out of their way to  change normal roads to cart paths?)

**Comment by Woazboat** [comment]
There, I made a proof of concept pull request for you: https://github.com/chadrockey/TGC-Designer-Tools/pull/119

**Comment by BrianZ111** [comment]
I forked and edited the code some time back for my own personal use to allow use of an OSM file instead of live data on the OSM servers.  I did this primarily for no longer existing courses and future courses not yet built - things that don’t belong on a live OSM map.  I’ve gone ahead and created a release out of those changes so others can use it:

https://github.com/BrianZ111/TGC-Designer-Tools/releases

I’m not sure how much it will actually be used by the users though.  First, in the documentation I encouraged people to continue to upload to the OSM live server for projects that align with OSM’s greater mapping uses because I believe it adds to the greater good of both OSM and the golf simulation community.  Second, there’s a lack of instruction on how to draw and save an OSM file in the community and I won’t have time to go into detail.  I’ve pointed out options such as JOSM and other CAD/GIS tools in the documentation but it will be up to the user to research how to do this or it will fall to someone else to create a more in depth tutorial.

**Comment by Woazboat** [comment]
That's great @BrianZ111 ! 
@chadrockey unfortunately seems to have abandoned this project, so would you be willing to merge the changes in https://github.com/chadrockey/TGC-Designer-Tools/pull/119 into your version? We could then point users to your version instead. 
You're probably right that it won't solve the problem of people creating non-existing golf courses without good instructions on how to do that properly, but it would help with other issues. Golf course mappers unfortunately also have a tendency to (incorrectly) change the tagging of roads just so this tool will pick them up. The linked PR could hopefully fix that by including more kinds of OSM roads by default.

**Comment by Woazboat** [comment]
Then I'm glad to be proven wrong. I based my assumption on the fact that this project hasn't seen any changes in more than two and a half years and that a lot of comments and issues seem to be left with no response (including this one for a week).

Issues #115 and #116 seem to be completely unrelated to the proposed changes in https://github.com/chadrockey/TGC-Designer-Tools/pull/119. I don't quite understand why they would block all other changes?

**Comment by chadrockey** [comment]
This is a 3rd party project released for free based around a commercial for profit video game, it’s not entirely up to me to gift their profit margins.  The last release of the game that applied to my system was in 2018/2019.  There’s a difference between needing to be updated and working relatively well for everyone until recently.

Most reported issues are user requests for assistance with data or operation and not true bugs.

I’m not going to do another release until I can address the very serious issue with the free satellite imagery source from Mapquest being taken offline.  Maybe I can find another source, maybe I cannot, but the project is functionally paused without that data source.

I don’t think you’ll have to worry about your roads until then because the app essentially does not work.

**Comment by Woazboat** [comment]
I understand that this is a project you're doing in your free time. Please keep in mind that OSM is also an open community project and contributors aren't getting paid for their work either.

I am unfortunately not so sure that the non-functional satellite imagery is an effective deterrent for people adding incorrect data to OSM for the purposes of exporting it via this tool. Here's a recent example from 5 days ago where a user incorrectly added an old golf course that no longer exists. They explicitly said that they were doing it so they could export the data to a golf simulator. After indignant comments towards other mappers who pointed out that what they were doing was incorrect, they left with the parting words "go ahead and delete it if you dont like it. I have the data I need.".

https://www.openstreetmap.org/changeset/132671033

This unfortunately also illustrates the big problems with these users. They only care about getting the data they need for their golf simulator and don't really care about their impact on OSM or if they are banned afterwards. The only way to combat this is to remove the incentives for adding incorrect data in the first place. That is what https://github.com/chadrockey/TGC-Designer-Tools/pull/119 attempts to do to prevent users from mapping incorrect road types. By extending the used OSM features to include other roads, there is no more need to change them to specific (incorrect) road types. 
@BrianZ111 's changes that allow offline editing without having to change the live data on the OSM servers would be a countermeasure against people adding nonexistent golf courses.

**Comment by chadrockey** [comment]
The is illustrates a problem in that the two communities don’t understand each other.  The satellite imagery is NOT used to add OSM data and if you had used the software before believing you can fix the malignant user behavior better than everyone here, you’d have also realized that.  Your edit won’t work and you’ll get stranger changes like users converting roads into closed splines, but I have the intention of merging it if I can resolve the other issues.  This is a limitation of the game, NOT of anything open source.

As I’ve recommended before to everyone else from OSM, you should permanently IP ban users who hurt your community and purge all of their edits.  They make the edits on OSM’s terms with OSM’s tools.  Make an example of them if you their data doesn’t help you.

**Comment by Woazboat** [comment]
Banning them does not work like I just mentioned. These are individual users whose only edits in OSM are to one or two golf courses they want to use in the game. Then they vanish and never think about OSM again. 

Of course OSM tools (and the satellite imagery available in these tools) are used to edit OSM and not the satellite imagery that is (no longer) available in this tool. That's not what I was saying. The point is that people are still using this tool, even without working satellite imagery
> I don’t think you’ll have to worry about your roads until then because the app essentially does not work.

**Comment by BrianZ111** [comment]
@Woazboat, the intention of my fork was merely to provide others a way to use modifications I originally only made for my own personal use.  I do not want to become the new main line support for the tool.  The only reason I did a release was because a user who knew I had done this but couldn’t figure out how to run the source in Python asked me to.  I happened upon this thread shortly after that and thought I’d drop a note here.   I’m happy either way if @chadrockey wants to include my changes at some point or not, but I don't really want to get into the business of adding other people's changes to my fork.  I wouldn't have time to support issues arising from changes I didn't make.
